src/methods.jl

```diff
@@ -179,3 +179,66 @@ function trimview(frame::AbstractArray, idxs)
 
     return selectdim(frame, d, complement_idxs)
 end
+
+
+"""
+    crop(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+Shape speficies the size of the cropped array, force_equal when set to true forces the cropped rows and columns to be equal on both sides
+otherwise the cropped frame contains one extra column on left and one extra row on top if necessary.
+
+# Examples
+```jldoctest
+julia> frame = reshape(1:25, (5, 5));
+
+julia> crop(frame, (3, 3))
+3×3 Array{Int64,2}:
+ 7  12  17
+ 8  13  18
+ 9  14  19
+
+```
+
+# See Also
+* [`cropview`](@ref)
+"""
+crop(frame::AbstractArray, shape; kwargs...) = copy(cropview(frame, shape; kwargs...))
+
+
+"""
+    cropview(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+This function is same as the [`crop`](@ref) function but returns a view of the frame.
+
+!!! note
+    This function returns a view of the frame, so any modification to output
+    array will result in modification of frame.
+
+# See Also
+* [`crop`](@ref)
+"""
+function cropview(frame::AbstractArray, shape; force_equal = true)
+    # testing error
+    ndims(frame) == length(shape) || error("Dimension mismatch between frame and shape")
+    any(s -> !isa(s, Colon) && s < 1, shape) && error("crop size $shape cant't be less than 1")
+
+    # generating idxs for cropped frame
+    idxs = map(enumerate(size(frame)), shape) do (d, s1), s2
+                diff = s2 isa Colon ? 0 : s1 - s2
+                lower = iseven(diff) ? (1 + (diff ÷ 2)) : (1 + ((diff - 1) ÷ 2))
+                upper = if isodd(diff) && force_equal
+                            @warn "dimension $d changed from $s2 to $(s2 + 1)"
+                            (s1 - ((diff - 1) ÷ 2))
+                        else
+                            iseven(diff) ? (s1 - (diff ÷ 2)) : (s1 - ((diff + 1) ÷ 2))
+                        end
+                lower:upper
+            end
```

Comment:
```suggestion
    idxs = map(enumerate(size(frame)), shape) do (d, s1), s2
                diff = s2 isa Colon ? 0 : s1 - s2
                lower = iseven(diff) ? diff ÷ 2 : (diff - 1) ÷ 2
                upper = if isodd(diff) && force_equal
                            @warn "dimension $d changed from $s2 to $(s2 + 1)"
                            (diff - 1) ÷ 2
                        elseif isodd(diff)
                            (diff + 1) ÷ 2
                        else
                            diff ÷ 2
                        end
                1 + lower:s1 - upper
            end
```

just a little easier to parse.